pkg/quota/admission/clusterresourcequota/admission.go

```diff
@@ -0,0 +1,132 @@
+package clusterresourcequota
+
+import (
+	"errors"
+	"io"
+	"sync"
+	"time"
+
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/quota"
+	"k8s.io/kubernetes/pkg/quota/install"
+	utilwait "k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/plugin/pkg/admission/resourcequota"
+
+	oclient "github.com/openshift/origin/pkg/client"
+	ocache "github.com/openshift/origin/pkg/client/cache"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	"github.com/openshift/origin/pkg/controller/shared"
+	"github.com/openshift/origin/pkg/quota/controller/clusterquotamapping"
+)
+
+func init() {
+	admission.RegisterPlugin("ClusterResourceQuota",
+		func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+			registry := install.NewRegistry(client)
+			// TODO: expose a stop channel in admission factory
+			return NewClusterResourceQuota(registry)
+		})
+}
+
+// quotaAdmission implements an admission controller that can enforce quota constraints
+type quotaAdmission struct {
+	*admission.Handler
+
+	// these are used to create the accessor
+	quotaLister     *ocache.IndexerToClusterResourceQuotaLister
```

Comment:
all the naming requests from earlier apply here as well since we are now dealing with two types of quota intermittently throughout the code.